src/reads-stdin.c

```diff
@@ -0,0 +1,68 @@
+#include <stdio.h>
+#include <string.h>
+
+// This program reads stdin and asserts that it is the expected value repeated
+// until stdin ends
+
+// Intentionally a prime length to make hitting buffering edge cases more likely
+const char expected_repeat[] = {'e', 'x', 'a', 'm', 'p', 'l', 'e'};
+
+int main(int argc, char** argv)
+{
+    size_t consumed_prefix = 0;
```

Comment:
What about:
```suggestion
    char buffer[20];
    const char expected[26] = "exampleexampleexampleexamp";
    size_t offset = 0; // always between 0 and 6
    for (;;)
    {
        size_t read_amount = fread(buffer, 1, sizeof(buffer), stdin);
        if (argc > 1) puts(argv[1]);
        if (read_amount == 0) {
            if (feof(stdin)) { puts("success"); return 0; }
            return 1;
        }
        if (memcmp(buffer, expected + offset, read_amount) != 0) return 2;
        offset = (offset + read_amount) % 7;
    }
```

Reply:
:O `const char expected[26] = "exampleexampleexampleexamp";` sometimes I wish I had thought of this....